gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjureBasePlugin.java

```diff
@@ -93,6 +94,11 @@ private static void createIrTasks(
             compileIr.getProductDependencies().set(project.provider(pdepsExtension::getProductDependencies));
             compileIr.dependsOn(copyConjureSourcesTask);
             compileIr.dependsOn(extractCompilerTask);
+            
+            project.getConfigurations().maybeCreate(CONJURE_IR_CONFIGURATION);
+            project.artifacts(artifactHandler -> {
+                artifactHandler.add(CONJURE_IR_CONFIGURATION, compileIr.getOutputIrFile());
```

Comment:
make sure this knows the `compileIr` task need to run to produce this. I think you need an extra param like
```
artifact -> artifact.builtBy(compileIr)
```